bokehjs/src/lib/models/plots/plot_canvas.ts

```diff
@@ -319,10 +319,14 @@ export class PlotView extends LayoutDOMView implements Paintable {
     }
 
     const {toolbar_location, toolbar_inner, toolbar} = this.model
-    if (toolbar_location != null) {
-      this._toolbar = new ToolbarPanel({toolbar})
-      toolbar.location = toolbar_location
-      toolbar.inner = toolbar_inner
+    this._toolbar_panel = new ToolbarPanel({toolbar})
+
+    const toolbar_visible = toolbar_location != null
+    this._toolbar_panel.visible = toolbar_visible
+    this._toolbar_panel.toolbar.inner = toolbar_inner
+
+    if (toolbar_visible) {
```

Comment:
Does this if-check even need to be here?

Reply:
`toolbar_location` can be `null` (see line 324), whereas `location` cannot. Note that `toolbar_visible` boolean constant carries nullablility constraint on `toolbar_location`.